src/components/remote_control/src/message_helper.cc

```diff
@@ -0,0 +1,154 @@
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string>
+#include "remote_control/message_helper.h"
+#include "remote_control/rc_module_constants.h"
+#include "utils/make_shared.h"
+
+namespace remote_control {
+using functional_modules::RCFunctionID;
+namespace {
+std::map<RCFunctionID, std::string> GenerateAPINames() {
+  std::map<RCFunctionID, std::string> result;
+  result.insert(std::make_pair<RCFunctionID, std::string>(
+      RCFunctionID::BUTTON_PRESS, "ButtonPress"));
+  result.insert(std::make_pair<RCFunctionID, std::string>(
+      RCFunctionID::GET_INTERIOR_VEHICLE_DATA, "GetInteriorVehicleData"));
+  result.insert(std::make_pair<RCFunctionID, std::string>(
+      RCFunctionID::SET_INTERIOR_VEHICLE_DATA, "SetInteriorVehicleData"));
+  result.insert(std::make_pair<RCFunctionID, std::string>(
+      RCFunctionID::ON_INTERIOR_VEHICLE_DATA, "OnInteriorVehicleData"));
+  result.insert(std::make_pair<RCFunctionID, std::string>(
+      RCFunctionID::ON_REMOTE_CONTROL_SETTINGS, "OnRemoteControlSettings"));
+  return result;
+}
+
+std::map<std::string, hmi_apis::Common_RCAccessMode::eType> access_modes{
+    {enums_value::kAutoAllow, hmi_apis::Common_RCAccessMode::AUTO_ALLOW},
+    {enums_value::kAutoDeny, hmi_apis::Common_RCAccessMode::AUTO_DENY},
+    {enums_value::kAskDriver, hmi_apis::Common_RCAccessMode::ASK_DRIVER}};
+}
+
+const std::map<RCFunctionID, std::string> MessageHelper::kMobileAPINames =
+    GenerateAPINames();
+
+const std::string MessageHelper::GetMobileAPIName(RCFunctionID func_id) {
+  std::map<RCFunctionID, std::string>::const_iterator it =
+      kMobileAPINames.find(func_id);
+  if (kMobileAPINames.end() != it) {
+    return it->second;
+  } else {
+    return "";
+  }
+}
+
+std::string MessageHelper::ValueToString(const Json::Value& value) {
+  Json::FastWriter writer;
+
+  return writer.write(value);
+}
+
+Json::Value MessageHelper::StringToValue(const std::string& string) {
+  Json::Reader reader;
+  Json::Value json;
+
+  return reader.parse(string, json) ? json
+                                    : Json::Value(Json::ValueType::nullValue);
+}
+
+bool IsMember(const Json::Value& value, const std::string& key) {
+  if (!value.isObject()) {
+    return false;
+  }
+
+  return value.isMember(key);
+}
+
+application_manager::MessagePtr MessageHelper::CreateHmiRequest(
+    const char* function_id,
+    const uint32_t hmi_app_id,
+    const Json::Value& message_params,
+    RemotePluginInterface& rc_module) {
+  using namespace json_keys;
+  Json::Value msg;
+
+  msg[json_keys::kId] = rc_module.service()->GetNextCorrelationID();
+
+  msg[kJsonrpc] = "2.0";
```

Comment:
Please assign explicit scope to the enums used in this file.

Reply:
@JackLivio  assigned scope explicitly